Assets/MixedRealityToolkit/MSBuild/Scripts/ReferenceItemInfo.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// A common base class for reference items such as C# Projects and DLLs to be added to MSBuild.
+    /// </summary>
+    public class ReferenceItemInfo
+    {
+        protected readonly IEnumerable<CompilationPlatformInfo> availablePlatforms;
+
+        /// <summary>
+        /// Gets the Guid associated with the reference.
+        /// </summary>
+        public Guid Guid { get; }
+
+        /// <summary>
+        /// Gets the output path to the reference.
+        /// </summary>
+        public Uri ReferencePath { get; }
+
+        /// <summary>
+        /// Gets name of the reference item.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// </summary>
+        /// <remarks>
+        /// In the editor, we can support all patforms if it's a pre-defined assembly, or an asmdef with Editor platform checked. 
```

Comment:
>In the editor, we can support all patforms if it's a pre-defined assembly, or an asmdef with Editor platform checked.  [](start = 12, length = 118)

I'm trying to understand this comment a little more - is this saying that if if the overall thing is being built for editor + assembly, or editor + asmdef, this gets ignored? i.e. so the usage of this is done somewhat externally of this class?

Reply:
Actually, this might actually be a sin of documentation, but it might be helpful just to point to:

private ReadOnlyDictionary<BuildTarget, CompilationPlatformInfo> GetCompilationPlatforms(bool inEditor)

Which is what defines the rules here.

---
In reply to: [301791189](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301791189) [](ancestors = 301791189)